components/rhs_thread/rhs_thread.tsx

```diff
@@ -32,40 +35,53 @@ export function renderThumbHorizontal() {
     return (<div/>);
 }
 
-export function renderThumbVertical(props) {
+export function renderThumbVertical(props: Props) {
     return (
         <div
             {...props}
             className='scrollbar--vertical'
         />);
 }
 
-export default class RhsThread extends React.Component {
-    static propTypes = {
-        posts: PropTypes.arrayOf(PropTypes.object).isRequired,
-        channel: PropTypes.object.isRequired,
-        selected: PropTypes.object.isRequired,
-        previousRhsState: PropTypes.string,
-        currentUserId: PropTypes.string.isRequired,
-        previewCollapsed: PropTypes.string.isRequired,
-        previewEnabled: PropTypes.bool.isRequired,
-        socketConnectionStatus: PropTypes.bool.isRequired,
-        actions: PropTypes.shape({
-            removePost: PropTypes.func.isRequired,
-            selectPostCard: PropTypes.func.isRequired,
-            getPostThread: PropTypes.func.isRequired,
-        }).isRequired,
-    }
+type Props = {
+    posts: Post[];
+    channel: Partial<Channel> | null;
+    selected: Record<string, any>;
+    previousRhsState?: string;
+    currentUserId: string;
+    previewCollapsed: string;
+    previewEnabled: boolean;
+    socketConnectionStatus: boolean;
+    actions: {
+        removePost: (post: ExtendedPost) => (dispatch: DispatchFunc, getState: GetStateFunc) => void;
+        selectPostCard: (post: Record<string, any>) => Record<string, any>;
```

Comment:
Similarly, this won't return anything because `selectPostCard` has also been wrapped by `bindActionCreators`. The original version of it returns a plain object as a redux action, but once that's wrapped, it doesn't return anything. Also, the `post` parameter is a `Post`, so the correct type for this should be:
```suggestion
        selectPostCard: (post: Post) => void;
```